### DISCUSSION TITLE
How to use last_modified hook

### DISCUSSION THREAD

**Discussion by groundedtako** [kungfux/kungfux.github.io#6] [Q&A] [answered]
#!/usr/bin/env ruby
#
# Check for changed posts

Jekyll::Hooks.register :posts, :post_init do |post|

  commit_num = `git rev-list --count HEAD "#{ post.path }"`

  if commit_num.to_i > 1
    lastmod_date = `git log -1 --pretty="%ad" --date=iso "#{ post.path }"`
    post.data['last_modified_at'] = lastmod_date
  end

end

Hi Alexander,
You added this hook to your chirpy template. I see that all your posts now have the last_modified_at timestamp.

How is this hook used? How do I also update my posts to have this timestamp? And also how is this timestamp used on the static site?

**Comment by kungfux** [comment, marked as answer]
Hey @groundedtako 

The last modified hook is a [part](https://github.com/cotes2020/chirpy-starter/blob/main/_plugins/posts-lastmod-hook.rb) of the Chirpy Starter. It checks the post history in Git to set the `post.data['last_modified_at']` value. So when you make changes to the post and commit changes to Git, the post will show up as modified.

![image](https://github.com/kungfux/kungfux.github.io/assets/10548881/63f0b7b3-9121-4823-8a62-b5051b2b4120)

The hook does not modify or add the `last_modified_at` in the front matter for a post, but you can add it manually. If `last_modified_at` is defined in the front matter, then the hook has no effect. So if I define it manually in the front matter, I have manual control over the last modified date.

```markdown
---
date: 2024-06-23 07:33:12 +0400
last_modified_at: 2024-06-23 07:33:12 +0400
---
```
![image](https://github.com/kungfux/kungfux.github.io/assets/10548881/e759e6d8-f9aa-4036-aa88-4d63badaa3bf)


Here is why I do this. I don't want the post to be marked as modified when I fix a typo and commit changes, but I prefer to show the post as modified when I decide to change the content, like adding or removing a paragraph. So basically, I avoid using the hook while it is still in the codebase.
